filament/src/Scene.cpp

```diff
@@ -199,6 +199,12 @@ void FScene::updateUBOs(utils::Range<uint32_t> visibleRenderables, backend::Hand
         mat3f m = mat3f::getTransformForNormals(model.upperLeft());
         m *= mat3f(1.0f / std::sqrt(max(float3{length2(m[0]), length2(m[1]), length2(m[2])})));
 
+        // Since our "getTransformForNormals" matrix omits the divide-by-determinant step,
+        // it has the wrong sign when the determinant is negative.
+        if (sceneData.elementAt<REVERSED_WINDING_ORDER>(i)) {
```

Comment:
Should we do this in `getTransformForNormals`?

Reply:
Maybe, but if we move it then we will be computing the determinant twice, which is somewhat redundant.  I'm on the fence.  @pixelflinger : you made the original switch to cof(), what do you think?

Reply:
Also note that `getTransformForNormals` is used in a couple other places, e.g. when determining dominant light direction.  So yeah maybe we should make it slower and just add a divide-by-det in there.

Reply:
Dale Weiler's [GitHub post](https://github.com/graphitemaster/normals_revisited) makes the argument that we should just use the raw cofactor matrix, but I didn't understand one of the steps in his algebra.  Moreover Lengyel never makes such a claim.  In practice, dividing the cof by the det (and therefore using the inverse-transpose) seems to fix this particular issue.